File: src/clang/fibo.c
#include <stdio.h>

int main() {
    int pre=1, preBeforePre=0, number;
    printf ("please input number:");
    scanf("%d", &number);
    int count = 0,result=0, total=1;;
    while (count < number) {
        result = pre + preBeforePre;
        total += result;
        preBeforePre = pre;
        pre = result;
        count ++;
    }
    printf("fibo result is %d and calculation result is %d\n", result, total);
    return 0;
}

File: src/go/algorithm/fibo.go
package main
import (
  "fmt"
  "io"
  "os"
)

func main() {
  count, firstEle, secondEle, currentEle, result := 0, 0, 1, 0, 0
  result = firstEle + secondEle
  fmt.Println("please input fibo number:")
  _, err  := fmt.Scan(&count)
  _ = err
  for i := 0; i < count; i++ {
    currentEle = firstEle + secondEle
    fmt.Println(currentEle)
    result += currentEle
    firstEle = secondEle
    secondEle = currentEle
    _ = result
    _ = firstEle
    _ = secondEle
  }
  s := fmt.Sprintln("fibo result is", currentEle, "and calculation result is", result)
  io.WriteString(os.Stdout, s)
}

